bootstrap/src/test.rs

```diff
@@ -53,56 +66,108 @@ impl TestCommand {
     pub fn collect_testcases(&self, manifest: &Manifest) -> Vec<TestCase> {
         let mut result = vec![];
 
+        // Test auxiliary (should compile first)
+        for case in glob("tests/auxiliary/*.rs").unwrap() {
+            let case = case.unwrap();
+            let filename = case.file_stem().unwrap();
+            let name = format!("auxiliary/{}", filename.to_string_lossy());
+            let output_file = manifest.out_dir.join(filename);
+            result.push(TestCase { name, source: case, output_file, test: TestType::CompileLib })
+        }
```

Comment:
Are aux crates just compiled for everything?

In rust-lang/rust we use a pattern like `//@ aux-build:mini_core.rs` ([example](https://github.com/rust-lang/rust/blob/b13176595d8a9fd27334747080764a53a18f4c24/tests/codegen/nounwind.rs#L1)). That would probably be worthwhile here to have better control over what is built / linked.

Just search the test files with regex `^//@\s*aux-build:(?P<fname>.*)`.

Reply:
I see you updated this, looks great